ui/src/message_popup/hint_card.jsx

```diff
@@ -0,0 +1,141 @@
+// @flow
+import React from 'react';
+import * as PropTypes from '../prop_types.js';
+import RaisedButton from 'material-ui/RaisedButton';
+import Divider from 'material-ui/Divider';
+
+/*
+Shows a card for a student from the virtual school.
+*/
+export default React.createClass({
+  displayName: 'HintCard',
+
+  getInitialState: function(){
+    return ({
+      hidden: true,
+      exampleIndex: 0,
+      nonExampleIndex: 0,
+    });
+  },
+  
+  propTypes: {
+    examples: React.PropTypes.array.isRequired,
+    nonExamples: React.PropTypes.array.isRequired
+  },
+  
+  onNextExample(){
+    var length = this.props.examples.length;
+    var next = this.state.exampleIndex + 1;
+    if(next >= length){
+      next = 0;
+    }
+    this.setState({exampleIndex: next});
+  },
+  
+  onNextNonExample(){
+    var length = this.props.nonExamples.length;
+    var next = this.state.nonExampleIndex + 1;
+    if(next >= length){
+      next = 0;
+    }
+    this.setState({nonExampleIndex: next});
+  },
+  
+  onHintsToggled(){
+    this.setState({ hidden: !this.state.hidden });
+  },
+
+  render() {
```

Comment:
This looks awesome!  Here's a design suggestion.  I agree there's too much going on if we show all the examples and non-examples at once.  How about the button "Show hint" shows one hint, which is either an example or non-example.  And then the "Next hint" becomes "Another hint" and when that's click it shows another example or non-example.

Let me know if it's easier to chat in person too!